src/AWS.jl

```diff
@@ -426,17 +456,40 @@ end
 (service::JSONService)(operation, args=[]) = service(aws_config(), operation, args)
 (service::JSONService)(a...; b...) = service(a..., b)
 
-function (service::RestJSONService)(aws::AWS.AWSConfig, request_method::String, request_uri::String, args=[])
-    return AWSCore.service_rest_json(
-        aws;
+function (service::RestJSONService)(
+    aws::AWS.AWSConfig,
+    request_method::String,
+    request_uri::String,
+    args::AbstractDict{String, <:Any},
+)
+    return_headers = _return_headers(args)
+
+    request = Request(
         service=service.name,
-        version=service.api_version,
-        verb=request_method,
-        resource=request_uri,
-        args=args
+        api_version=service.api_version,
+        request_method=request_method,
+        headers=LittleDict{String, String}(get(args, "headers", [])),
+        resource = _generate_rest_resource(request_uri, args),
```

Comment:
Keep things consistent
```suggestion
        resource=_generate_rest_resource(request_uri, args),
```